sql/group_test.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Peter Mattis (peter@cockroachlabs.com)
+
+package sql
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/util/leaktest"
+)
+
+func TestDesiredAggregateOrder(t *testing.T) {
+	defer leaktest.AfterTest(t)
+
+	testData := []struct {
+		expr     string
+		ordering []int
+	}{
+		{`MIN(a)`, []int{1}},
```

Comment:
`parseAndNormalizeExpr` can't handle an empty expression because it does so by trying to parse `SELECT` which is not a valid statement. Instead, I added a case for `a` which does not contain an aggregation function which is what I think you meant to test. I added the other test cases as well. Note that `MIN(3)` is normalized to `3` which is then not of the render targets for the scan node when aggregation is being performed.